assets/js/util.js

```diff
@@ -242,8 +243,19 @@ function synchronizeTextareaHeights() {
     $('#translatedText').css('height', originalTextScrollHeight + 'px');
 }
 
-/*:: export {synchronizeTextareaHeights, modeEnabled, ajaxSend, ajaxComplete, filterLangList, onlyUnique,
-    SPACE_KEY_CODE, ENTER_KEY_CODE, HTTP_OK_CODE, HTTP_BAD_REQUEST_CODE, XHR_LOADING, XHR_DONE} */
+function callApy(options, endpoint) {
+    return $.ajax(Object.assign({
+        url: config.APY_URL + endpoint,
+        beforeSend: ajaxSend,
+        contentType: 'application/x-www-form-urlencoded; charset=UTF-8',
+        dataType: 'json',
+        type: (options.data.q.length > THRESHOLD_REQUEST_LENGTH) ? 'POST' : 'GET'
```

Comment:
@sushain97 , I read through what `jsonp` means and why it works only `GET`. Are you suggesting something like the following?

```
function callApy(options, endpoint) {
    var callApyObject = Object.assign({
        url: config.APY_URL + endpoint,
        beforeSend: ajaxSend,
        contentType: 'application/x-www-form-urlencoded; charset=UTF-8'
    }, options);

    if(options.data.q.length > THRESHOLD_REQUEST_LENGTH) {
        callApyObject.type = 'POST';
        callApyObject.dataType = 'json';
        return $.ajax(callApyObject);
    }
    else {
        callApyObject.type = 'GET';
        callApyObject.dataType = 'jsonp';
        return $.jsonp(callApyObject);
    }
}
```